app/src/main/java/lt/vilnius/tvarkau/views/adapters/ProblemImagesPagerAdapter.java

```diff
@@ -0,0 +1,53 @@
+package lt.vilnius.tvarkau.views.adapters;
+
+import android.content.Context;
+import android.support.v4.view.PagerAdapter;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+
+import lt.vilnius.tvarkau.R;
+
+/**
+ * Created by Gediminas Zukas on 2016-04-28.
+ */
+public class ProblemImagesPagerAdapter extends PagerAdapter {
+
+    private LayoutInflater mLayoutInflater;
+    private int[] mResources;
+
+    public ProblemImagesPagerAdapter(Context context, int[] imagesIds) {
+        mLayoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        mResources = imagesIds;
+    }
+
+    @Override
+    public int getCount() {
+        return mResources.length;
+    }
+
+    @Override
+    public boolean isViewFromObject(View view, Object object) {
+        return view == object;
+    }
+
+    @Override
+    public Object instantiateItem(ViewGroup container, int position) {
+        // TODO: consider recycling views
```

Comment:
It's pretty easy to implement view recycling in PagerAdapter. For now it will work perfectly without it. So let's consider implementing it in the future :)